src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfAnalyzer.cs

```diff
@@ -126,10 +126,21 @@ void PopulateLocalReferencesSet(OperationAnalysisContext context)
                         }
 
                         var parent = localReference.Parent;
-                        if (!(parent is IBinaryOperation binaryOperation))
+                        if (parent is IBinaryOperation binaryOperation)
                         {
-                            localsToBailOut.Add(localReference.Local);
+                            var operatorKind = binaryOperation.OperatorKind;
+                            var otherOperand = binaryOperation.LeftOperand is ILocalReferenceOperation ? binaryOperation.RightOperand : binaryOperation.LeftOperand;
+                            if (otherOperand.ConstantValue.HasValue && otherOperand.ConstantValue.Value is int intValue)
+                            {
+                                if ((operatorKind == BinaryOperatorKind.Equals && intValue < 0) ||
+                                    (operatorKind == BinaryOperatorKind.GreaterThanOrEqual && intValue == 0))
+                                {
+                                    // Do nothing. This is a valid case to the tagged in the analyzer
+                                    return;
+                                }
+                            }
```

Comment:
Can we add a static helper method for this code that can be reused even below?

Reply:
Yup.